content/blog/2019-10-23_react-typescript-starter.md

```diff
@@ -0,0 +1,464 @@
++++
+title = "A boilerplate for a React+TypeScript frontend: part 1"
+description = "How to quickly get started with a React/TypeScript project with Webpack"
++++
+
+Starting a new frontend project is tiresome. My usual process is to look at the last project I worked on, 
+try to extract all the base dependencies and configuration and see if a Hello World still works after upgrading everything.
+This problem might be alleviated with tools such as [create-react-app](https://github.com/facebook/create-react-app) but
+I personally prefer not having an abstraction layer for the configuration as it is something I always end up having to
+tweak. Even when using frameworks like [Next.js](https://nextjs.org/) where the configuration is hidden, you still need
+to make some changes and understanding [Webpack](https://webpack.js.org/) goes a long way in the frontend world.
+
+The article is split in two parts: 
+
+1. Getting a local dev environment set up: this article
+2. Making it production ready: an upcoming article
+
+The focus for these article is to set up a build process for modern browsers with TypeScript: I am not going to pick any library other than React.
+State management, testing framework etc are all up to you. The slight
+exception to that rule is that I will use [Sass](https://sass-lang.com/) to handle the styling rather than alternatives such as Less/CSS-in-JS.
+Once again, this is up to you. If you prefer to use JS for your CSS, just skip the last section of this article.
+
+I am also going to use [yarn](https://yarnpkg.com/lang/en/) as package manager but `npm` will work as well.
+
+You can view the end result in <https://github.com/Keats/webpack-react-typescript>.
+
+## TypeScript setup
+[TypeScript](https://www.typescriptlang.org/) is the most important tool for me in the frontend world. I first talked about 
+it on this blog [in 2014](@./blog/2014-5-4_switching-to-typescript.md) and it only got better with time. This is a superset of JavaScript, simply adding 
```

Comment:
Not that simple. I'm very confused as to when they decide to adopt a new es6 feature, and from what stage.

Reply:
TS implemente tous les proposals qui arrivent au stage 3 (eg acceptés). La seule exception c'est les décorateurs qui ont été implémentés avant.

Reply:
Je vois, merci!